services/task-svc/internal/task/task.go

```diff
@@ -132,6 +197,150 @@ func (ServiceServer) UpdateTask(ctx context.Context, req *pb.UpdateTaskRequest)
 	return &pb.UpdateTaskResponse{}, nil
 }
 
+func (ServiceServer) DeleteTask(ctx context.Context, req *pb.DeleteTaskRequest) (*pb.DeleteTaskResponse, error) {
+	log := zlog.Ctx(ctx)
+	db := hwgorm.GetDB(ctx)
+
+	// TODO: Auth
+
+	id, err := uuid.Parse(req.Id)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	task := Task{ID: id}
+
+	if err := db.Delete(&task).Error; err != nil {
+		log.Warn().Err(err).Msg("database error")
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	return &pb.DeleteTaskResponse{}, nil
+}
+
+func (ServiceServer) AddSubTask(ctx context.Context, req *pb.AddSubTaskRequest) (*pb.AddSubTaskResponse, error) {
+	log := zlog.Ctx(ctx)
+	db := hwgorm.GetDB(ctx)
+
+	// TODO: Auth
+
+	taskId, err := uuid.Parse(req.TaskId)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	// Check if task exists
+	if err := db.First(&Task{ID: taskId}).Error; err != nil {
+		if hwgorm.IsOurFault(err) {
+			return nil, status.Error(codes.Internal, err.Error())
+		} else {
+			return nil, status.Error(codes.InvalidArgument, "patientId not found")
+		}
+	}
+
+	done := false
+	if req.Done != nil {
+		done = *req.Done
+	}
+	subtask := Subtask{Title: req.Title, TaskID: taskId, Done: done}
```

Comment:
Is this the result of your `(req.Done = nil) != false` experiment?

Reply:
we have to check whether the pointer is a null pointer and set the default to false this accomplishes that or how would you solve this?